lib/oneview-sdk/image-streamer/resource/api300/golden_image.rb

```diff
@@ -10,13 +10,123 @@
 # language governing permissions and limitations under the License.
 
 require_relative 'resource'
+require 'net/http/post/multipart'
 
 module OneviewSDK
   module ImageStreamer
     module API300
       # Golden Image resource implementation for Image Streamer
       class GoldenImage < Resource
         BASE_URI = '/rest/golden-images'.freeze
+        READ_TIMEOUT = 300 # in seconds, 5 minutes
+        ACCEPTED_FORMATS = %w(.zip .ZIP).freeze # Supported upload extensions
+
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::ImageStreamer::Client] client The client object for the Image Streamer appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          super
+          # Default values:
+          @data['type'] ||= 'GoldenImage'
+        end
+
+        # Download the details of the golden image capture logs which has been archived based on the specific attribute ID.
+        # @param [String] file_path
+        # @return [True] When was saved successfully
+        def get_details_archive(file_path)
+          ensure_client && ensure_uri
+          resp = @client.rest_api(:get, "#{BASE_URI}/archive/#{@data['uri'].split('/').last}")
+          File.open(file_path, 'wb') { |file| file.write(resp.body) }
+          true
+        end
+
+        # Downloads the content of the selected golden image as per the specified attributes.
+        # @param [String] file_path
+        # @return [True] When was saved successfully
+        def download(file_path)
+          ensure_client && ensure_uri
+          uri = URI.parse(URI.escape("#{client.url}#{BASE_URI}/download/#{@data['uri'].split('/').last}"))
+          req = Net::HTTP::Get.new(uri.request_uri)
+
+          options = {}
+          options['Content-Type'] = 'application/json'
+          options['X-Api-Version'] = @client.api_version.to_s
+          options['auth'] = @client.token
+          options.each do |key, val|
+            req[key] = val
+          end
+
+          http_request = Net::HTTP.new(uri.host, uri.port)
+          http_request.use_ssl = true
+          http_request.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
Also, just curious, will this ever be a long enough request that we'll want to set the read_timeout like we do below?